### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by balanalina**
Demo Video:
https://drive.google.com/file/d/1mZ41oTr-o1RfmKREixPEqnVKqPjS7Vi8/view?usp=sharing

### CODE REVIEW

File: css-scss/mainPage.css [hunk 1]
```diff
@@ -0,0 +1,291 @@
+.topContainer {
```
Comment (florinionce) [inline css-scss/mainPage.css:1]: the css selectors should use `kebab-case` (`top-container`)

File: css-scss/mainPage.css [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+  background: url(caffee2.jpg) no-repeat center fixed;
+  height: 550px;
+  background-size: cover;
+  width: 100%;
+  z-index: 1;
+  position: relative;
+  text-align: center;
+}
```
Comment (florinionce) [inline css-scss/mainPage.css:9]: you should leave an empty row after these blocks. You have on some of them, so now it's not consistent :D

File: css-scss/mainPage.scss [hunk 3]
```diff
@@ -0,0 +1,338 @@
+$font: 'Averia Serif Libre';
+$fontSize: 15px;
+
+.topContainer{
+
```
Comment (florinionce) [inline css-scss/mainPage.scss:5]: extra space :D

File: css-scss/mainPage.scss [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+	background: url(caffee2.jpg) no-repeat center fixed;
+	height: 550px;
+	background-size: cover;
+	width: 100%;
+	z-index: 1;
+	position: relative;
+	text-align: center;
+	
+	.caffeeLogo{
```
Comment (florinionce) [inline css-scss/mainPage.scss:14]: you should add a space before the `{`

File: css-scss/mainPage.scss [hunk 5]
```diff
@@ -0,0 +1,338 @@
...
+		display: inline-block;
+		padding-top: 2.7%;
+	}
+
+	nav{
+		width: 100%;
+		//background-color: green;
```
Comment (florinionce) [inline css-scss/mainPage.scss:21]: you can remove the commented code

File: css-scss/mainPage.scss [hunk 6]
```diff
@@ -0,0 +1,338 @@
...
+			width: auto;
+			margin-left: 12%;
+			margin-top: 2%;
+		}
+	}
+
+}
+
+.rightSideInfo{
+	//display: inherit;
```
Comment (florinionce) [inline css-scss/mainPage.scss:52]: same here

File: css-scss/mainPage.scss [hunk 7]
```diff
@@ -0,0 +1,338 @@
...
+		padding-right: 20px;
+		padding-left: 20px;
+		&:hover{
+			color: orange;
+		}
+	}
+}
+
+.dot{
+	height: 7px;
```
Comment (florinionce) [inline css-scss/mainPage.scss:72]: 👀

File: css-scss/mainPage.scss [hunk 8]
```diff
@@ -0,0 +1,338 @@
...
+ 	img{
+ 		//padding: 10px;
+ 		display: inline-block;
+ 		margin-top: 6px;
+ 		height: 25px;
+ 	}
+
+ }
+
+ .randomText{
```
Comment (florinionce) [inline css-scss/mainPage.scss:116]: this is probably not a suggestive name

File: css-scss/mainPage.scss [hunk 9]
```diff
@@ -0,0 +1,338 @@
...
+ 	//margin-left: 80px;
+ 	font-family: $font;
+ 	font-size: $fontSize +5;
```
Comment (florinionce) [inline css-scss/mainPage.scss:119]: `$fontsize + 5` :D mind the spaces

File: css-scss/mainPage.scss [hunk 10]
```diff
@@ -0,0 +1,338 @@
...
+		align-items: flex-start;
+		//margin: 1px;
+		background-color: white;
+		transition: all .2s ease-in-out;
+
+		&:hover {
+          transform: scale(1.08);
+          z-index: 10; 
+        }
+
```
Comment (florinionce) [inline css-scss/mainPage.scss:245]: the indentation seems broken

File: js/mainPage.js [hunk 11]
```diff
@@ -0,0 +1,226 @@
+
+function loadSlider(){
+	//create the slider
+	var slider = document.getElementById('mySlider');
+	noUiSlider.create(slider, {
+    	start: [0, 100],
```
Comment (florinionce) [inline js/mainPage.js:6]: the indentation seems wrong here

File: js/mainPage.js [hunk 12]
```diff
@@ -0,0 +1,226 @@
...
+	    }
+	});
+	//event for the display of the slider range
+	slider.noUiSlider.on('update',function(values,handler){
+	document.getElementById("first").innerHTML="$"+values[0];
+	document.getElementById("second").innerHTML="$"+values[1];
+	$('.menu').innerHTML = "";	
+	loadElements(parseInt(values[0]),parseInt(values[1]));
+});
+	var ok=localStorage.getItem("itemsLength");
```
Comment (florinionce) [inline js/mainPage.js:20]: what does `ok` stand for? also the indentation seems wrong

Comment (florinionce) [reply js/mainPage.js:20]: Is that variable used?

File: js/mainPage.js [hunk 13]
```diff
@@ -0,0 +1,226 @@
...
+}
+//sets slider values to 0 a and 100
+function setSlider(){
+	var slider = document.getElementById('mySlider');
+	slider.noUiSlider.set([0, 100]);
+}
+//display the div with add/remove operations
+$(document).on('click','.item',function(){
+		$(".delAdd").fadeIn().delay(7000).fadeOut();
+		var clas = $(this).attr('class');
```
Comment (florinionce) [inline js/mainPage.js:34]: this variable name is not very suggestive :D

File: js/mainPage.js [hunk 14]
```diff
@@ -0,0 +1,226 @@
...
+		$(".del").click(function(){ deleteEl(nr); });
+	});
+
+//takes input and adds a new element
+function addEl(){
+	var title = prompt("Enter the name of the dish: ","Gourmet Croissant");
+	var category = prompt("Enter the cateogory of the dish: ","Breakfast");
+	var price = prompt("Enter the price of teh dish: ","$12");
+	var src = prompt("Enter the dish picture source: ","");
+	console.log(title);
```
Comment (florinionce) [inline js/mainPage.js:45]: you forgot this

File: js/mainPage.js [hunk 15]
```diff
@@ -0,0 +1,226 @@
...
+$(document).on('click','.item',function(){
+		$(".delAdd").fadeIn().delay(7000).fadeOut();
+		var clas = $(this).attr('class');
+		nr = clas.slice(5);
+		$(".del").click(function(){ deleteEl(nr); });
+	});
+
+//takes input and adds a new element
+function addEl(){
+	var title = prompt("Enter the name of the dish: ","Gourmet Croissant");
```
Comment (florinionce) [inline js/mainPage.js:41]: it's probably better to leave a space after the `,` just like you would do in a sentence :D

File: js/mainPage.js [hunk 16]
```diff
@@ -0,0 +1,226 @@
...
+	var price = prompt("Enter the price of teh dish: ","$12");
+	var src = prompt("Enter the dish picture source: ","");
+	console.log(title);
+	if(title == null || category == null || price == null || src == null)
+		return;
+	if(title.length == 0 || category.length == 0 || price.length == 0 || src.length == 0)
+		return;
+	var length = parseInt(localStorage.getItem("itemsLength"));
+	length+=1;
+	localStorage.setItem('itemsLength',length);
```
Comment (florinionce) [inline js/mainPage.js:52]: I'm wondering why you need this. Can't you count the elements in the local storage?  You could store them in an array. like this: 
```
  dishes: [
    { 
         category: 'pasta',
         name: '....',
     },
     {
      .....
     }
  ]
```

File: js/mainPage.js [hunk 17]
```diff
@@ -0,0 +1,226 @@
...
+	localStorage.setItem('src'+length,src);
+	var length = parseInt(localStorage.getItem('size'));
+	localStorage.setItem('length',length+1);
+	//set the slider to the default values
+	setSlider();
+	//reoad teh elements
+	loadElements(0,100);
+}
+
+
```
Comment (florinionce) [inline js/mainPage.js:65]: you have extra spaces here

File: js/mainPage.js [hunk 18]
```diff
@@ -0,0 +1,226 @@
...
+	if(currentElement%4==0)
+		return "right";
+	if(nrOfElements%4!=0){
+		if(currentElement%4==1)
+			return "leftbottom";
+		if(currentElement%4==2)
+			return "bottom";
+		if(currentElement%4==3)
+			return "bottom";
+	}
```
Comment (florinionce) [inline js/mainPage.js:147]: please mind the indentation

File: js/mainPage.js [hunk 19]
```diff
@@ -0,0 +1,226 @@
...
+
+//load the elements by filter
+function loadElements(lowerBound,upperBound){
+	var length = localStorage.getItem('itemsLength');;
+	//put a div for every row
+	//load the add/remove div
+	var el="<div class='delAdd'><p class='del'>Delete</p>"+
+			"<p onclick='addEl()'>Add</p></div>";
+			var count =0;
```
Comment (florinionce) [inline js/mainPage.js:157]: 👀

File: js/mainPage.js [hunk 20]
```diff
@@ -0,0 +1,226 @@
...
+		if(count%4==0)
+			el+="</div>"	
+
+
+	}
+	//load the divs
+	$('.menu').html(el);
+	//set the id for the outline 
+	/*
+	$('.1').attr("id","lefttop");
```
Comment (florinionce) [inline js/mainPage.js:184]: you can remove the commented code

File: mainPage.html [hunk 21]
```diff
@@ -2,18 +2,187 @@
...
+				<p>PRICE</p>
+				<p class="price" id="first"></p>
+				<div id="mySlider"></div>
+				<!--<input type="range" name="">-->
+				<p class="price" id="second"></p>
 			</div>
 		</nav>
 	</div>
+	<div class="menu">
+		<!--static dishes-->
```
Comment (florinionce) [inline mainPage.html:80]: you can remove the commented code :D

**Review by florinionce — COMMENTED** [review]
Nice. It looks good overall, but please be more careful with the indentation. Also, the node modules folder should be placed in gitignore :D

File: js/mainPage.js (hunk 12 above)
Comment (balanalina) [reply js/mainPage.js:20]: It is a "leftover" from something else, I thought that it was deleted 😓

**Pull request merged**